### DISCUSSION TITLE
Is there a way to enable RLS for all tables automatically?

### DISCUSSION THREAD

**Discussion by andrewrjohn** [supabase/supabase#12264] [Questions]
Pardon my ignorance if this is a dumb question, but I don't understand why RLS isn't enabled by default for all newly created tables. I am using Supabase Auth in my app and Prisma as my DB client to connect to my Supabase instance. It would be really nice if RLS was enabled by default, since my tables are "managed" by Prisma.

If this isn't possible, is there another way I could get around this in order to make all tables locked down by default?

**Comment by GaryAustin1** [comment]
There is no current setting to make RLS default.

You can take anon (and if need be authenticated) roles away from the public schema...
https://github.com/supabase/supabase/discussions/1493#discussioncomment-724842
That also shows it has been debated on default access to anon or not by the SB team.
I think there was another way by changing the ability of authenticator role to not have anon/authenticated roles, but I can't find it here in a quick search.
